### PR TITLE
Add trivial bash fixes

### PR DESCRIPTION
**Pull request by matejak**
Implement trivial fixes mainly related to
- quoting
- unused variables
- bash shebangs that cause problems if they are present in our source snippets - the build system may not preserve them on the first line of the bash remediation content.

### CODE REVIEW

File: shared/macros-bash.jinja [hunk 1]
```diff
@@ -260,7 +260,7 @@ do
 
                 # Split concrete_rule into head, perm, and tail sections using hash '#' delimiter
                 rule_head=$(cut -d '#' -f 1 <<< "$concrete_rule")
-                rule_perm=$(cut -d '#' -f 2 <<< "$concrete_rule")
+                # rule_perm=$(cut -d '#' -f 2 <<< "$concrete_rule")  # not needed in the remediation, but that's what is stored there
```
Comment (yuumasato) [inline shared/macros-bash.jinja:263]: I think you can remove the unused `rule_perm` variable and update the comment in line 261.

File: linux_os/guide/services/sssd/sssd_certificate_verification/bash/shared.sh [hunk 2]
```diff
@@ -22,9 +22,9 @@ done
 
 if ! $found ; then
 	SSSD_CONF="/etc/sssd/conf.d/certificate_verification.conf"
-	mkdir -p $( dirname $SSSD_CONF )
-	touch $SSSD_CONF
-	chown root:root $SSSD_CONF
-	chmod 600 $SSSD_CONF
-	echo -e "[sssd]\ncertificate_verification = ocsp_dgst = $var_sssd_certificate_verification_digest_function" >> $SSSD_CONF
+	mkdir -p "$( dirname "$SSSD_CONF" )"
```
Comment (yuumasato) [inline linux_os/guide/services/sssd/sssd_certificate_verification/bash/shared.sh:25]: Are there any guidance about spaces after `(` and before `)`?

File: linux_os/guide/services/sssd/sssd_run_as_sssd_user/bash/shared.sh [hunk 3]
```diff
@@ -16,9 +16,9 @@ done
 
 if ! $found ; then
 	SSSD_CONF="/etc/sssd/conf.d/ospp.conf"
-	mkdir -p $( dirname $SSSD_CONF )
-	touch $SSSD_CONF
-	chown root:root $SSSD_CONF
-	chmod 600 $SSSD_CONF
-	echo -e "[sssd]\nuser = sssd" >> $SSSD_CONF
+	mkdir -p "$( dirname "$SSSD_CONF" )"
```
Comment (yuumasato) [inline linux_os/guide/services/sssd/sssd_run_as_sssd_user/bash/shared.sh:19]: Same thing here about the spaces after and before `(` and `)`.

**Review by yuumasato — CHANGES_REQUESTED** [review]
Thanks for these fixes.

File: linux_os/guide/services/sssd/sssd_certificate_verification/bash/shared.sh (hunk 2 above)
Comment (matejak) [reply linux_os/guide/services/sssd/sssd_certificate_verification/bash/shared.sh]: none that I am aware of, so removing those.

**Pull request merged**